### ISSUE TITLE
VPP MCTF is disabled in default library configuration build

### ISSUE THREAD

**Issue by dvrogozh** [Intel-Media-SDK/MediaSDK#144]
Here is how MCTF is enabled in the mediasdk library:
https://github.com/Intel-Media-SDK/MediaSDK/blob/0e50c533fc57f06af703ed41614d170972ed10a5/_studio/shared/include/mfx_config.h#L216

Hence, there is no chance to get feature workable with the default build which should match API=1.26 at the moment:
`perl tools/builder/build_mfx.pl --cmake=intel64.make.release`

**Comment by changliwu** [comment]
I modify   #define MFX_VERSION_NEXT (MFX_VERSION_MAJOR * 1000 + MFX_VERSION_MINOR + 1) to #define MFX_VERSION_NEXT (MFX_VERSION_MAJOR * 1000 + MFX_VERSION_MINOR in the api/include/mfxdefs.h, then the MCTF is enabled in the mediasdk library, But when i run ./sample_multi_transcode -i::h264 input.264 -o::h264 out.264 -mctf 10 and ./sample_multi_transcode -i::h264 input.264 -o::h264 out.264, the result of out.264 can't find any difference

**Comment by mmishuro** [comment]
1) -mctf  option needs to be given in par file
2) same trick with MFX_VERSION_NEXT needs to be done for samples (there is
also configuration file)
finally, check help info that is printed by sample_multi_transcode (or
sample_vpp) if mctf is enabled, it should print help info with MCTF details.

good luck, Mike

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by changliwu** [comment]
2) same trick with MFX_VERSION_NEXT needs to be done for samples (there is also configuration file)

which configuration file can i change for the value of MFX_VERSION_NEXT？

thanks！

**Comment by onabiull** [comment]
To enable pre-release functionality you should generate makefiles with following command:
```sh
perl tools/builder/build_mfx.pl --cmake=intel64.make.release --api=latest
```
But since commit [e2b1ca61](https://github.com/Intel-Media-SDK/MediaSDK/commit/e2b1ca61fb2e69ac3e200c2597787dd3099eedfa) MCTF should be enabled even without it.

**Comment by changliwu** [comment]
Thank you for your answer！
I have anther question, I run the demo of sample_multi_transcode with the API 1.26 of MediaSDK,
like this:
./sample_multi_transcode -i::h264 input.264 -o::h264 out.264 -cqp -qpi 24 -qpp 26 -qpb 31 -denoise 10
./sample_multi_transcode -i::h264 input.264 -o::h264 out.264 -cqp -qpi 24 -qpp 26 -qpb 31
but the size of the two output files(out.264) has little change, Is the denoise is  enabled in API 1.26 ?

**Comment by dvrogozh** [comment]
This was addressed with e2b1ca61fb2e69ac3e200c2597787dd3099eedfa
